### PR TITLE
[chore]: enable int-conversion rule of perfsprint

### PR DESCRIPTION
**Pull request by mmorel-35**
#### Description

[perfsprint](https://github.com/catenacyber/perfsprint) is a linter for performance, aiming at usages of fmt.Sprintf which have faster alternatives.

This PR enables int-conversion rule

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -154,6 +155,12 @@ linters-settings:
     locale: US
     ignore-words:
       - cancelled
+  perfsprint:
```
Comment (MrAlias) [inline .golangci.yml:158]: Can this be configured to not check test files?

**Review by pellared — APPROVED** [review]

File: .golangci.yml [hunk 2]
```diff
@@ -154,6 +155,12 @@ linters-settings:
...
+    err-error: false
+    errorf: false
+    int-conversion: true
+    sprintf1: false
+    strconcat: false
```
Comment (dmathieu) [inline .golangci.yml:163]: Are you planning to enable those in further PRs? (I think we should)

**Review by dmathieu — APPROVED** [review]

File: .golangci.yml (hunk 1 above)
Comment (MrAlias) [reply .golangci.yml:158]: Performance is not critical for tests and it provides an alternate way to test in a functionally equivalent way.

Comment (pellared) [reply .golangci.yml:158]: @mmorel-35 I think you can simply add `perfsprint` to https://github.com/open-telemetry/opentelemetry-go/blob/0d48e02ad81af1feb5d6e6ceeb70e0a05c0326d6/.golangci.yml#L65-L68

**Pull request merged**